### DISCUSSION TITLE
[Exercise-2]: Write about variables (constants too) in your own words and share a sample PL/SQL block with compiled results.

### DISCUSSION THREAD

**Discussion by acrlakshman** [exaslate-courses/oracle-sql-plsql-b3#3] [General]
Write about normal variables and constants.

Ref.: https://www.youtube.com/watch?v=S0XnhFOmCJ0&list=PLF6_5RI6ENIlDi3knzcQGUiFc5-WdA_1W&index=1&t=1175s

**Comment by srideviadapa0511** [comment]
A variable is a name where we store values in the program temporarily for the execution of code. constants are same as variables but once we assign a value to a variable we can change or edit the value but when we declare a constant we cannot edit or change values stored in variables .

**Reply by acrlakshman** [reply]
Thanks for the comment @srideviadapa0511 . To summarize you mentioned, constant cannot be edited its value, but variable can be.

**Reply by acrlakshman** [reply]
@srideviadapa0511 Please also share screenshot of a sample *working* PL/SQL code with variable and constants in it.

**Comment by srideviadapa0511** [comment]
![image](https://github.com/exaslate-courses/oracle-sql-plsql-b3/assets/145452411/664b304a-1fec-4b24-8dac-5a625c9d3df0)

**Reply by acrlakshman** [reply]
@srideviadapa0511 Nice work.

**Comment by srideviadapa0511** [comment]
![STUDENT](https://github.com/exaslate-courses/oracle-sql-plsql-b3/assets/145452411/8d6e5342-4868-4f6f-b6d8-544f6c473245)

**Comment by YaminiYadavv** [comment]
Variables can be declared either in the declare section or Begin section, Variables are keywords that can be changed or reassigned to the code values and it can hold some of the datatypes like Integers, Strings and more. They are assigned using the symbol. (':=')
Whereas Constant keywords are considered as static in nature which cannot be changed once assigned throughout the code.
Variables are used to store data and considered as Dynamic in nature which means they can be reassigned to a program.
### Examples of Constant keywords:
![CONSTANT](https://github.com/exaslate-courses/oracle-sql-plsql-b3/assets/144749161/084507ad-7ad6-4a11-ada6-e2039b0e4980)
![image](https://github.com/exaslate-courses/oracle-sql-plsql-b3/assets/144749161/b45d0c5f-32cd-44b0-988d-5f04ea9c2209)
### Examples of Variables:
![variable](https://github.com/exaslate-courses/oracle-sql-plsql-b3/assets/144749161/ebc00fda-e972-4ca3-9518-7c0a0d59771a)

**Reply by acrlakshman** [reply]
@YaminiYadavv nice explanation and demos.

**Comment by Tharun1596** [comment]
variable is any value that can be store the data in declare section 
In variables the values  can be change in the code and where in constants cannot be change in the code

**Comment by Tharun1596** [comment]
![Capture](https://github.com/exaslate-courses/oracle-sql-plsql-b3/assets/141386177/79952005-678f-4192-9714-e5a1b162c536)

**Reply by acrlakshman** [reply]
@Tharun1596 please show the output also, i.e. I am expecting ` N : 10` as the output from your code.

**Comment by cm2024jsp** [comment]
DECLARE 
   a integer := 10; 
   b integer := 20; 
   c integer; 
   f real; 
DD CONSTANT NUMBER:=8.5;
BEGIN 
   c := a + b; 
   dbms_output.put_line('Value of c: ' || c); 
dbms_output.put_line('Value of D: ' || DD); 
   END;
